api-report/container-definitions.api.md

```diff
@@ -56,6 +56,7 @@ export type ConnectionState = ConnectionState.Disconnected | ConnectionState.Est
 
 // @public
 export enum ContainerErrorType {
+    accessViolationError = "accessViolationError",
```

Comment:
i'm sure you've though about it, but i find this error name to be really confusing.  to me access violation has to due with invalid memory access in a native application.

please think about how a consumer would think about this error name without any context on why it is occurring.

Reply:
i would personally probably just go with invalid operation

Reply:
https://stackoverflow.com/search?q=access+violation&s=4e2a161f-792d-4133-9489-10705df50546

Reply:
Thanks for the feedback, that's why I added you to the PR :)

I feel that this is analogous in the Fluid world. I think of AV coming up in native apps when you've messed up your pointers and memory management, e.g. you keep a pointer around to a block that's been freed.  That's basically what happens here - You keep a reference around to something that's otherwise unreferenced.

So yeah, it's intentionally borrowing on that concept.  I like that it comes with some baggage - people will hopefully understand it's a big deal.

Reply:
I could call it fluidAccessViolationError but we don't have "fluid" in other errorTypes.

Reply:
I think it's fine to call it fluidAccessViolationError. It makes sense that it's a fluid problem

Reply:
i still feel its a stretch and mix of metaphors that feels more clever than useful. Javascript, and most other languages with garbage collection don't have manual memory management, so don't have avs, so why would fluid, which is based on javascript have them.

Reply:
Maybe we ditch it after the request pattern goes away ;)

I'm totally open to something else, my concern with InvalidOperation is that it's too generic. I'd like something that is 1-1 with GC-related mis-steps

Reply:
i think you'll run into two kinds of people, people familiar with native code, who think they have a security issue, and people unfamiliar with native code who will think they have an auth problem, i don't think either will connect back to the real problem without lots of effort, and the misalignment expectations could actually slow that effort to understand down.

Reply:
Any other ideas besides InvalidOperation?

Reply:
DeletedObjectAccessed or DeletedObjectUsed?

Reply:
Did some brainstorming. Not necessarily looking for votes, but if any of these resonate with anyone, or inspire a better idea, lmk!

- Access Violation
- Invalid Operation
- Target Object Gone
- Gone Violation
- Fluid Object Gone

Reply:
This error is supposed to be temporary and should go away once we fully enable sweep. Basically, there would be no way to distinguish between deleted objects and objects that never existed. So, I would choose a name that very clearly tells what happened and not worry about the name being "official".

Reply:
I fall closer to the side of being unfamiliar with native code and AccessViolation did trip me up from a first glance. I also think InvalidOperation seems quite generic and could apply to other situations beyond referencing an unreferenced object. Will there be an accompanying message with these errors as well to give context to the error? From the current suggestions I can't say either InvalidOperation or AccessViolation will lead me to think immediately that it's an object reference problem.

What about 'Invalid Object Access'? or somerthing along the lines of an unreferenced object error?

Reply:
Invalid Object Reference?

Reply:
i assumed you were going for generic with access violation. if you are going for specific, just name it something very specific. what scenario or scenarios is this thrown in, ideally it points directly to what the dev did wrong.

Reply:
CannotRequestObjectDeletedByGargbageCollection

Reply:
my thought was we'd only add something generic, like InvalidOperation to the type, to avoid back/forward compat breaks, and then have a sub-status/reason like above that is very specific.

Reply:
Right, if we had an accompanying message/sub-status/reason. I think I would opt for the more generic variants that have been suggested. I don't know which one I would prefer in that situation.

If we are aiming to have this error type be specific to this scenario - I think Invalid Object Reference is great! It also covers the future scenarios (or even current?) where we cannot distinguish between a deleted/unreferenced object vs an object that never existed in the first place.

Reply:
Access Violation Error makes me think of authentication/authorization - if these issues are not intended to show up here, I'd prefer a name that doesn't make me think of those. CannotRequestObjectDeletedByGargbageCollection is perfectly clear in this regard if this error type is specific to this scenario :)

Reply:
Thanks for the input everyone! Going to go with a more specific and likely temporary name for now, and will defer updating the definitions too.